HeterogeneousCore/SonicCore/src/SonicClientBase.cc

```diff
@@ -57,7 +62,10 @@ void SonicClientBase::finish(bool success, std::exception_ptr eptr) {
     edm::LogInfo(fullDebugName_) << "Client time: "
                                  << std::chrono::duration_cast<std::chrono::microseconds>(t1 - t0_).count();
   }
-  holder_.doneWaiting(eptr);
+  if (holder_)
+    holder_->doneWaiting(eptr);
+  else if (eptr)
```

Comment:
How about
```suggestion
  if (holder_) {
    holder_->doneWaiting(eptr);
    holder_.reset();
  } else if (eptr)
```
to make it clear that the old `WaitingTaskWithArenaHolder` won't be used e.g. by any (future) mistake?